x/authenticator/authenticator/message_filter.go

```diff
@@ -125,24 +192,18 @@ func isSuperset(a, b interface{}) error {
 		}
 
 	case string:
-		if bv, ok := b.(string); ok {
-			// Attempt to treat strings as numbers if they look like numbers
-			if decA, err := sdk.NewDecFromStr(av); err == nil {
```

Comment:
The reason for this is that numbers in messages are often encoded as strings but this may not be clear to the person creating the pattern. If your pattern is `{"amount": 10}` and the message *has* amount=10. The matching would fail (without this code) becasuse `10 != "10"`

Reply:
Reverted to the old impl in the latest commit